js/os.ts

```diff
@@ -23,6 +23,41 @@ interface CodeInfo {
   sourceMap: string | undefined;
 }
 
+interface IsTTY {
+  stdin: boolean;
+  stdout: boolean;
+  stderr: boolean;
+}
+
+let _isTTY: IsTTY; // cache
+
+/** Check if running in terminal.
+ *
+ * isTTY().stdin; // stdin
+ * isTTY().stdout; // stdout
+ * isTTY().stderr; // stderr
+ */
+export function isTTY(): IsTTY {
+  if (_isTTY) {
+    return _isTTY; // return cached value if we have one
+  }
+
+  const builder = flatbuffers.createBuilder();
+  msg.IsTTY.startIsTTY(builder);
+  const inner = msg.IsTTY.endIsTTY(builder);
+  const baseRes = sendSync(builder, msg.Any.IsTTY, inner)!;
+  assert(msg.Any.IsTTYRes === baseRes.innerType());
+  const res = new msg.IsTTYRes();
+  assert(baseRes.inner(res) != null);
+
+  _isTTY = {
+    stdin: res.stdin(),
+    stdout: res.stdout(),
+    stderr: res.stderr()
+  };
+  return _isTTY;
```

Comment:
Don't cache the value, just return it.

Reply:
@ry why? @hayd said that we need cache.

Reply:
Why doesn't it make sense to cache?

Reply:
@hayd I think so, it can't be updated, if we run a program in TTY, it'll be in TTY

Reply:
The cache may introduce subtle bugs - and there are zero performance implications. Just do the simple thing.